runtime/src/bank.rs

```diff
@@ -1491,19 +1496,21 @@ impl Bank {
             vote_only_bank: false,
             cost_tracker: RwLock::<CostTracker>::default(),
             sysvar_cache: RwLock::<SysvarCache>::default(),
-            accounts_data_len: AtomicU64::default(),
+            accounts_data_size_initial: 0,
+            accounts_data_size_delta_on_chain: AtomicI64::new(0),
+            accounts_data_size_delta_off_chain: AtomicI64::new(0),
             fee_structure: FeeStructure::default(),
         };
 
-        let accounts_data_len = bank.get_total_accounts_stats().unwrap().data_len as u64;
-        if accounts_data_len != 0 {
-            bank.store_accounts_data_len(accounts_data_len);
-
-            let cost_tracker = CostTracker::new_with_account_data_size_limit(
-                bank.feature_set
-                    .is_active(&feature_set::cap_accounts_data_len::id())
-                    .then(|| MAX_ACCOUNTS_DATA_LEN.saturating_sub(accounts_data_len)),
-            );
+        let accounts_data_size_initial = bank.get_total_accounts_stats().unwrap().data_len as u64;
+        bank.accounts_data_size_initial = accounts_data_size_initial;
```

Comment:
Is there a reason to replace a setter (and I see you deleted the old setter fn) with just setting the value directly?

Reply:
Hmm, maybe GitHub changed where the highlighted part of the code was? One thing that may be at play here, is that because I split the initial size and the delta, I do not require interior mutability for the `accounts_data_size_initial`, so this field is basically a constant after the new bank is created. Previously, `accounts_data_len` was an Atomic that could be changed ay any point, so I made the `store_accounts_data_len()` fn to ensure the proper atomic ordering was used.